### PR TITLE
Add AMG private endpoint commands

### PR DESCRIPTION
**Pull request by Jeralan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az grafana

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

Adds private endpoint commands to azure managed grafana extension. Generated via aaz-dev-tools

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/amg/HISTORY.rst [hunk 1]
```diff
@@ -84,3 +84,8 @@ Release History
 * `az grafana list`: Migrate to AAZDev Tool
 * `az grafana show`: Migrate to AAZDev Tool
 * `az grafana delete`: Migrate to AAZDev Tool
+
+2.1.0
+++++++
+* `az grafana private-endpoint-connection`: support private endpoint connection management
+* `az grafana managed-private-endpoint`: support managed private endpoint management
```
Comment (ABZhang0) [inline src/amg/HISTORY.rst:91]: let's shorten this to 'az grafana mpe'

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py [hunk 2]
```diff
@@ -440,6 +440,89 @@ def test_amg_backup_restore(self, resource_group):
...
+            'sub': self.get_subscription_id(),
+        })
+
+        owner = self._get_signed_in_user()
+        self.recording_processors.append(MSGraphNameReplacer(owner, MOCKED_USER_NAME))
+
+        with unittest.mock.patch('azext_amg.custom._gen_guid', side_effect=self.create_guid):
+
+            self.cmd('grafana create -g {rg} -n {name} -l {location}')
+            time.sleep(120)
```
Comment (ABZhang0) [inline src/amg/azext_amg/tests/latest/test_amg_scenario.py:464]: is this delay necessary? We didn't use it for creation in the test_amg_crud() scenario

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py [hunk 3]
```diff
@@ -440,6 +440,89 @@ def test_amg_backup_restore(self, resource_group):
...
+            'name': 'clitestamgmpe',
+            'location': 'westcentralus',
+            'monitor_name': 'clitestamgmonitor',
+            'mpe_name': 'clitestmpe',
+            'pe_name': 'clitestpe',
+            'vnet_name': 'clitestvnet',
+            'subnet_name': 'clitestsubnet',
+            'conn_name': 'clitestconn',
+            'sub': self.get_subscription_id(),
+        })
```
Comment (ABZhang0) [inline src/amg/azext_amg/tests/latest/test_amg_scenario.py:456]: consider using random names to avoid collisions: https://github.com/Azure/azure-cli/blob/master/doc/authoring_tests.md#sample-6-random-name-and-name-mapping

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py [hunk 4]
```diff
@@ -440,6 +440,89 @@ def test_amg_backup_restore(self, resource_group):
...
+                'mpe_id': out[0]['id']
+            })
+
+            self.cmd('network private-endpoint-connection approve --id {mpe_id} -d "Approved" ')
+
+            self.cmd('grafana managed-private-endpoint show -g {rg} --workspace-name {name} -n {mpe_name}', checks=[
+                self.check('connectionState.status', 'Pending')
+            ])
+
+            self.cmd('grafana managed-private-endpoint refresh -g {rg} --workspace-name {name}')
```
Comment (ABZhang0) [inline src/amg/azext_amg/tests/latest/test_amg_scenario.py:492]: any possible checks we can do for the 'refresh' command?

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py [hunk 5]
```diff
@@ -440,6 +440,89 @@ def test_amg_backup_restore(self, resource_group):
...
+                self.check('connectionState.status', 'Approved')
+            ])
+
+            self.cmd('grafana managed-private-endpoint delete -g {rg} --workspace-name {name} -n {mpe_name} --yes')
+
+            self.cmd('grafana managed-private-endpoint list -g {rg} --workspace-name {name}', checks=[
+                self.check('length([])', 0)
+            ])
+
+            self.cmd('network vnet create -g {rg} -n {vnet_name} -l {location} --subnet-name {subnet_name}')
```
Comment (ABZhang0) [inline src/amg/azext_amg/tests/latest/test_amg_scenario.py:504]: any possible checks we can do after creation to check that the resource attributes are what we expect?

**Review by ABZhang0 — CHANGES_REQUESTED** [review]

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py (hunk 4 above)
Comment (Jeralan) [reply src/amg/azext_amg/tests/latest/test_amg_scenario.py]: Not directly on the call to the refresh command. The refresh command does not output anything. We check that the connection state updates from the refresh though by checking before and after.

File: src/amg/azext_amg/tests/latest/test_amg_scenario.py (hunk 5 above)
Comment (Jeralan) [reply src/amg/azext_amg/tests/latest/test_amg_scenario.py:512]: Added checks to the mpe creation, the other create calls are creating non-amg resources.

**Comment by Jeralan** [comment]
@yonzhan ready for review

**Review by kairu-ms — APPROVED** [review]
